analyzers/src/SonarAnalyzer.CSharp/Rules/NotAssignedPrivateMember.cs

```diff
@@ -87,56 +82,65 @@ protected override void Initialize(SonarAnalysisContext context)
                     var usedMemberSymbols = new HashSet<ISymbol>(usedMembers.Select(tuple => tuple.Symbol));
 
                     var assignedMemberSymbols = GetAssignedMemberSymbols(usedMembers);
+                    var unassignedUsedMemberSymbols = allCandidateMembers.Where(x => usedMemberSymbols.Contains(x.Symbol) && !assignedMemberSymbols.Contains(x.Symbol));
 
-                    foreach (var candidateMember in allCandidateMembers)
+                    foreach (var candidateMember in unassignedUsedMemberSymbols)
                     {
-                        if (!usedMemberSymbols.Contains(candidateMember.Symbol))
-                        {
-                            /// reported by <see cref="UnusedPrivateMember"/>
-                            continue;
-                        }
-
-                        if (!assignedMemberSymbols.Contains(candidateMember.Symbol))
-                        {
-                            var field = candidateMember.SyntaxNode as VariableDeclaratorSyntax;
-                            var property = candidateMember.SyntaxNode as PropertyDeclarationSyntax;
+                        var field = candidateMember.SyntaxNode as VariableDeclaratorSyntax;
+                        var property = candidateMember.SyntaxNode as PropertyDeclarationSyntax;
 
-                            var memberType = field != null ? "field" : "auto-property";
+                        var memberType = field != null ? "field" : "auto-property";
```

Comment:
we should invert the condition
```suggestion
                        var memberType = field == null ? "auto-property" : "field";
```